### PR TITLE
support multi-platform container image builds

### PR DESCRIPTION
**Pull request by nickytd**
/area logging

This PR improves logging project Makefile. 

- It reduce dependencies to external binaries.
- `test` targets now uses `go install ...`to fetch linter and test dependencies
- container image build target now uses `buildx` plugin unified with project CI setup
- `REGISTRY` and `DOCKER_BUILD_PLATFORM` can now be suppled as env variables. 

```other developer
This PR aligns container build targets with project CI supporting multi-platform builds and simplifies overall Makefile structure.
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -58,47 +71,74 @@ install-copy:
...
 
 .PHONY: revendor
 revendor:
 	@GO111MODULE=on go mod tidy
 	@GO111MODULE=on go mod vendor
 
 .PHONY: check
-check:	
-	@chmod +x $(REPO_ROOT)/vendor/github.com/gardener/gardener/hack/check.sh
-	@$(REPO_ROOT)/vendor/github.com/gardener/gardener/hack/check.sh --golangci-lint-config=./.golangci.yaml ./cmd/... ./pkg/...
```
Comment (vlvasilev) [inline Makefile:88]: Why we are breaking the chain with gardener `hack` scripts?

File: hack/docker-image-build.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
```
Comment (vlvasilev) [inline hack/docker-image-build.sh:3]: Copyright (c) 2022 SAP ...

File: hack/docker-image-build.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+# limitations under the License.
+
+set -e
+dir=$(dirname $0)
+
+echo "> Docker build"
+
+TARGET="${1:-}"
+if [ -z $TARGET ]; then
+  echo "TARGET is a required parameter"
```
Comment (vlvasilev) [inline hack/docker-image-build.sh:24]: I am not very experienced with bash but if something is required and it is not present then we should exit the script?

File: hack/docker-image-build.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  echo "IMAGE_TAG is a required parameter"
+fi
+
+EFFECTIVE_VERSION="${4:-}"
+
+pushd $dir/..
+
+docker build \
+  --build-arg EFFECTIVE_VERSION="${EFFECTIVE_VERSION}" \
+  --tag "${IMAGE_REPOSITORY}:${IMAGE_TAG}-latest" \
```
Comment (vlvasilev) [inline hack/docker-image-build.sh:43]: Why we are using `${IMAGE_TAG}-latest` instead of `latest`?

File: hack/docker-image-push.sh [hunk 5]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
```
Comment (vlvasilev) [inline hack/docker-image-push.sh:3]: Copyright (c) 2022 ...

File: hack/docker-image-push.sh [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+	echo "creating $BUILDER builder"
+	docker buildx create --name $BUILDER --use
+fi
+
+pushd $dir/..
+docker buildx build --push --platform=$DOCKER_BUILD_PLATFORM \
+  --build-arg EFFECTIVE_VERSION="${EFFECTIVE_VERSION}" \
+  --tag "${IMAGE_REPOSITORY}:${IMAGE_TAG}-latest" \
+	--tag "${IMAGE_REPOSITORY}:${IMAGE_TAG}" \
+	-f Dockerfile --target ${TARGET} .
```
Comment (vlvasilev) [inline hack/docker-image-push.sh:58]: may be you should call `popd` here?

**Comment by nickytd** [comment]
The feedback is applied.

File: Makefile [hunk 7]
```diff
@@ -12,31 +12,44 @@
...
+VALI_CURATOR_IMAGE_REPOSITORY              := $(REGISTRY)/vali-curator
+TELEGRAF_IMAGE_REPOSITORY                  := $(REGISTRY)/telegraf-iptables
+TUNE2FS_IMAGE_REPOSITORY                   := $(REGISTRY)/tune2fs
+EVENT_LOGGER_IMAGE_REPOSITORY              := $(REGISTRY)/event-logger
+IMAGE_TAG                                  := $(VERSION)
+EFFECTIVE_VERSION                          := $(VERSION)-$(shell git rev-parse HEAD)
+
+# project folder structure
+PKG_DIR                                    := $(REPO_ROOT)/pkg
+TOOLS_DIR                                  := $(REPO_ROOT)/hack
```
[the remark this answers is not in the document]
Comment (nickytd) [reply Makefile]: done

File: hack/docker-image-build.sh (hunk 4 above)
Comment (nickytd) [reply hack/docker-image-build.sh:43]: done

File: hack/docker-image-push.sh (hunk 5 above)
Comment (nickytd) [reply hack/docker-image-push.sh]: done

**Review by vlvasilev — APPROVED** [review]
/lgtm

**Pull request merged**